app/lib/shared/integrity.dart

```diff
@@ -426,50 +386,45 @@ class IntegrityChecker {
     }
   }
 
-  Future<void> _checkLikes() async {
+  Stream<String> _checkLikes() async* {
     _logger.info('Scanning Likes...');
 
-    await for (Like like in _db.query<Like>().run()) {
+    await for (final like in _db.query<Like>().run()) {
       if (like.packageName == null) {
-        _problems.add(
-            'Like entity for user ${like.userId} and ${like.package} has a '
-            '`packageName` property which is not at string ');
+        yield 'Like entity for user "${like.userId}" and package "${like.package}" has a '
+            '`packageName` property which is not at string.';
       } else if (like.packageName != like.package) {
-        _problems.add('Like entity for user ${like.userId} and ${like.package}'
-            ' has a `packageName` property which is not the same as `package`/id');
+        yield 'Like entity for user "${like.userId}" and package "${like.package}"'
+            ' has a `packageName` property which is not the same as `package`/`id`.';
       }
 
-      _userToLikes.update(like.userId, (l) => l..add(like.package),
-          ifAbsent: () => <String?>[]);
-    }
-
-    _userToLikes.keys
-        .where((user) =>
-            (!_userToOauth.keys.contains(user) || _deletedUsers.contains(user)))
-        .forEach((user) {
-      _problems.add('Like entity with nonexisting or deleted user $user');
-    });
+      final userId = like.userId!;
+      if (!_userToOauth.keys.contains(userId) ||
+          _deletedUsers.contains(userId)) {
+        yield 'Like entity with nonexisting or deleted user "$userId".';
+      }
```

Comment:
We might want to split this in two cases to make the reporting precise.
```suggestion
      if (!_userToOauth.keys.contains(userId)) {
        yield 'Like entity with nonexisting user "$userId".';
      } else if (_deletedUsers.contains(userId)) {
         yield 'Like entity with deleted user "$userId".';
      }
```